### PR TITLE
[WIP] Implementation of avl_ndel()

### PR DESCRIPTION
**Pull request by matthiasbeyer**
Added implementation of avl_ndel() with helper function, which is not implemented at this moment, as #143 and #144 are not done yet, but required for this implementation.

### CODE REVIEW

File: src/libreset/avl.c [hunk 1]
```diff
@@ -226,6 +240,19 @@ avl_del(
...
 }
 
+unsigned int
+avl_ndel(
+    struct avl* el,
+    r_predf pred,
+    void* etc,
+    struct r_set_cfg* cfg
+) {
+    if (el && cfg) {
```
Comment (neithernut) [inline src/libreset/avl.c]: The `__nonull__` attribute should ensure that neither `el` nor `cfg` are NULL.

Comment (matthiasbeyer) [inline src/libreset/avl.c]: But you can ignore the compiler warnings, can't you?

Comment (neithernut) [inline src/libreset/avl.c]: Nope: `-Werror`

Comment (matthiasbeyer) [inline src/libreset/avl.c]: So I'll remove this check.

**Comment by neithernut** [comment]
If you rebase this branch onto master as soon as PR #148 is merged, I can complete the implementation for you.

**Comment by matthiasbeyer** [comment]
@neithernut Here you go.

**Comment by matthiasbeyer** [comment]
Solves #134

**Comment by neithernut** [comment]
I think this can be merged now.

**Comment by matthiasbeyer** [comment]
@neithernut Are you fine with rebasing and squashing your commits? (I will of course try to keep you as author).

**Comment by neithernut** [comment]
Yep.

**Comment by neithernut** [comment]
ACK for merge.

**Pull request merged**